custom-events/functions/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START import]
+const {onCustomEventPublished} = require("firebase-functions/v2/eventarc");
+const logger = require("firebase-functions/logger");
+const admin = require("firebase-admin");
+// [END import]
+
+admin.initializeApp();
+
+// [START imageresizedEvent]
+exports.onimageresized = onCustomEventPublished(
```

Comment:
I think it'd look nicer if we used came case for the exported function name.

@jhuleatt - do we have a style guide that mandates naming of exported functions?

Reply:
v2 doesn't support uppercase letters in function names, unfortunately :(

Reply:
Supported characters are lowercase letters, numbers, and dashes. However, dashes get tricky with JS. `exports.on-image-resized = ...` is invalid JS. I believe we can do `exports['on-image-resized']`, though.

Reply:
Yes, and the CLI will error with the following message if unsupported chars are used:

> v2 function name(s) can only contain lower case letters, numbers, hyphens, and not exceed 62 characters in length